src/TurbulenceConvection/EDMF_Environment.jl

```diff
@@ -0,0 +1,447 @@
+function microphysics(
+    ::SGSMean,
+    grid::Grid,
+    state::State,
+    edmf::EDMFModel,
+    precip_model::AbstractPrecipitationModel,
+    Δt::Real,
+    param_set::APS,
+)
+    FT = float_type(state)
+    thermo_params = TCP.thermodynamics_params(param_set)
+    tendencies_pr = center_tendencies_precipitation(state)
+    aux_en = center_aux_environment(state)
+    prog_pr = center_prog_precipitation(state)
+    prog_gm = center_prog_grid_mean(state)
+    aux_gm = center_aux_grid_mean(state)
+    ts_env = center_aux_environment(state).ts
+    p_c = aux_gm.p
+    ρ_c = prog_gm.ρ
+    aux_en_sat = aux_en.sat
+    aux_en_unsat = aux_en.unsat
+    precip_fraction = compute_precip_fraction(edmf, state)
+
+    @inbounds for k in real_center_indices(grid)
+        # condensation
+        ts = ts_env[k]
+        # autoconversion and accretion
+        mph = precipitation_formation(
+            param_set,
+            precip_model,
+            prog_pr.q_rai[k],
+            prog_pr.q_sno[k],
+            aux_en.area[k],
+            ρ_c[k],
+            FT(grid.zc[k].z),
+            Δt,
+            ts,
+            precip_fraction,
+        )
+
+        # update_sat_unsat
```

Comment:
Would be good to leave TODO comment here: We should clean the following up and remove the variables we do not need in a subsequent iteration (e.g., θ_liq_ice). It's also confusing that θ_virt is only defined for unsaturated condensations but not otherwise.